test/hpke_test.c

```diff
@@ -0,0 +1,1804 @@
+    OSSL_HPKE_SUITE stirred;
+    char giant[2048];
+    size_t suitesize;
+    size_t ptr_suitesize;
+
+    for (kemind = 0; kemind != OSSL_NELEM(kem_str_list); kemind++) {
+        for (kdfind = 0; kdfind != OSSL_NELEM(kdf_str_list); kdfind++) {
+            for (aeadind = 0; aeadind != OSSL_NELEM(aead_str_list); aeadind++) {
+                snprintf(sstr, 128, "%s,%s,%s", kem_str_list[kemind],
+                         kdf_str_list[kdfind], aead_str_list[aeadind]);
+                if (TEST_true(OSSL_HPKE_str2suite(sstr, &stirred)) != 1) {
+                    overallresult = 0;
+                }
+            }
+        }
+    }
+    for (sind = 0; sind != OSSL_NELEM(bogus_suite_strs); sind++) {
+        if (TEST_false(OSSL_HPKE_str2suite(bogus_suite_strs[sind],
+                                           &stirred)) != 1) {
+            overallresult = 0;
+        }
+    }
+    /* check a few errors */
+    if (!TEST_false(OSSL_HPKE_str2suite("", &stirred)))
+        overallresult = 0;
+    if (!TEST_false(OSSL_HPKE_str2suite(NULL, &stirred)))
+        overallresult = 0;
+    if (!TEST_false(OSSL_HPKE_str2suite("", NULL)))
+        overallresult = 0;
+    memset(giant, 'A', sizeof(giant) - 1);
+    giant[sizeof(giant) - 1] = '\0';
+    if (!TEST_false(OSSL_HPKE_str2suite(giant, &stirred)))
+        overallresult = 0;
+
+    /* we'll check the size of a suite just to see what we get */
+    suitesize=sizeof(stirred);
+    ptr_suitesize=sizeof(&stirred);
+    if (verbose) {
+        TEST_note("Size of OSSL_HPKE_SUITE is %d, size of ptr is %d",
+                  (int) suitesize, (int) ptr_suitesize);
+    }
+
+    return overallresult;
+}
+
+/**
+ * @brief try the various GREASEy APIs
+ * @return 1 for success, other otherwise
+ */
+static int test_hpke_grease(void)
+{
+    int overallresult = 1;
+    OSSL_HPKE_SUITE g_suite;
+    unsigned char g_pub[OSSL_HPKE_TSTSIZE];
+    size_t g_pub_len = OSSL_HPKE_TSTSIZE;
+    unsigned char g_cipher[OSSL_HPKE_TSTSIZE];
+    size_t g_cipher_len = 266;
+    size_t clearlen = 128;
+    size_t expanded = 0;
+    size_t enclen = 0;
+    size_t ikmelen = 0;
+
+    memset(&g_suite, 0, sizeof(OSSL_HPKE_SUITE));
+    /* GREASEing */
+    /* check too short for public value */
+    g_pub_len = 10;
+    if (TEST_false(OSSL_HPKE_get_grease_value(testctx, NULL, NULL, &g_suite,
+                                              g_pub, &g_pub_len,
+                                              g_cipher, g_cipher_len)) != 1) {
+        overallresult = 0;
+    }
+    /* reset to work */
+    g_pub_len = OSSL_HPKE_TSTSIZE;
+    if (TEST_true(OSSL_HPKE_get_grease_value(testctx, NULL, NULL, &g_suite,
+                                             g_pub, &g_pub_len,
+                                             g_cipher, g_cipher_len)) != 1) {
+        overallresult = 0;
+    }
+    /* expansion */
+    expanded = OSSL_HPKE_get_ciphertext_size(g_suite, clearlen);
+    if (!TEST_size_t_gt(expanded,clearlen)) {
```

Comment:
Nit: missing space after `,` - same comments applies twice more below here in this function